### PR TITLE
servicebus jms queue sample update

### PR DESCRIPTION
**Pull request by jialigit**
- [ ] Attention： remember to cherry pick to spring-cloud-azure_4.0.0-beta.4 branch

## Purpose

*Update the ServiceBus Jms Queue sample
1. Update configuration yml file.
2. Update README file.
3. Refactor code, including Filename, variable names...

## Does this introduce a breaking change?

```
[ ] Yes
[x ] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test
*  Get the code

```
git clone [repo-address]
cd [repo-name]
git checkout [branch-name]
npm install
```

* Test the code

```
```

## What to Check
Verify that the following are valid
* ...

### CODE REVIEW

**Comment by backwind1233** [comment]
@yiliuTo please help review this PR.

File: servicebus/spring-cloud-azure-starter-servicebus-jms/servicebus-jms-queue/src/main/java/com/azure/spring/sample/jms/queue/SyncQueueSendController.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    @PostMapping("/queue/sync")
+    public User postMessageAndReceive(@RequestParam String message) throws JMSException {
+        LOGGER.info("Sending message");
+        jmsTemplate.setReceiveTimeout(1000L);
+        jmsMessagingTemplate.setJmsTemplate(jmsTemplate);
+        Session session = jmsMessagingTemplate.getConnectionFactory().createConnection()
+                                              .createSession(false, Session.AUTO_ACKNOWLEDGE);
+        ObjectMessage objectMessage = session.createObjectMessage(new User(message));
+        objectMessage.setJMSCorrelationID(UUID.randomUUID().toString());
+        objectMessage.setJMSReplyTo(connectionFactory.createContext().createTemporaryQueue());
```
Comment (yiliuTo) [inline servicebus/spring-cloud-azure-starter-servicebus-jms/servicebus-jms-queue/src/main/java/com/azure/spring/sample/jms/queue/SyncQueueSendController.java:62]: what is the temporary queue here mean? will it create a queue in Service Bus? And we can comment here to introduce the meaning of message properties we configured.

File: servicebus/spring-cloud-azure-starter-servicebus-jms/servicebus-jms-queue/src/main/java/com/azure/spring/sample/jms/queue/SyncQueueSendController.java [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+        objectMessage.setJMSExpiration(1000L);
+        objectMessage.setJMSDeliveryMode(DeliveryMode.NON_PERSISTENT);
+
+        return jmsMessagingTemplate.convertSendAndReceive(new JmsQueue(queueName),
+            objectMessage, User.class); //this operation seems to be blocking + sync
```
Comment (yiliuTo) [inline servicebus/spring-cloud-azure-starter-servicebus-jms/servicebus-jms-queue/src/main/java/com/azure/spring/sample/jms/queue/SyncQueueSendController.java:67]: we shouldn't have "seems to be" here

File: servicebus/spring-cloud-azure-starter-servicebus-jms/servicebus-jms-queue/src/main/resources/application.yaml [hunk 3]
```diff
@@ -2,5 +2,8 @@ spring:
   jms:
     servicebus:
       connection-string: ${SERVICEBUS_NAMESPACE_CONNECTION_STRING}
-      idle-timeout: 1800000
+      idle-timeout: 5m
       pricing-tier: ${PRICING_TIER}
+      pool:
+        enabled: true # To be true to enable the JmsPoolConnectionFactory bean. Default false.
+queuename: que001
```
Comment (yiliuTo) [inline servicebus/spring-cloud-azure-starter-servicebus-jms/servicebus-jms-queue/src/main/resources/application.yaml:9]: Let's follow unified property naming schema and use kebab here

**Review by yiliuTo — CHANGES_REQUESTED** [review]

**Comment by yiliuTo** [comment]
We can update this sample and introduce the usage of caching and pooled connection factory

File: servicebus/spring-cloud-azure-starter-servicebus-jms/servicebus-jms-queue/README.md [hunk 4]
```diff
@@ -108,23 +108,30 @@ Running the command below to export environment values:
 
 ## Run Locally
 
-In your terminal, run `mvn clean spring-boot:run`.
-
-
+Navigate to the project root directory, then run the maven command:
```
Comment (backwind1233) [inline servicebus/spring-cloud-azure-starter-servicebus-jms/servicebus-jms-queue/README.md:111]: we can update the template and then update all the descriptions here.
issue [link](https://github.com/Azure-Samples/azure-spring-boot-samples/issues/149)

please revert it here.

File: servicebus/spring-cloud-azure-starter-servicebus-jms/servicebus-jms-queue/README.md [hunk 5]
```diff
@@ -108,23 +108,30 @@ Running the command below to export environment values:
...
 ```shell
 mvn clean spring-boot:run
 ```
 
 ## Verify This Sample
 
+1. Verify in your app's logs that a similar message was posted:
```
Comment (backwind1233) [inline servicebus/spring-cloud-azure-starter-servicebus-jms/servicebus-jms-queue/README.md:118]: similar messages were posted

File: servicebus/spring-cloud-azure-starter-servicebus-jms/servicebus-jms-queue/README.md [hunk 6]
```diff
@@ -108,23 +108,30 @@ Running the command below to export environment values:
...
+    ```
 
-1. Send a POST request to service bus queue.
+2. [Optional] Send a POST request to service bus queue.
     ```
     curl -d "" http://localhost:8080/queue?message=hello 
     ```
 
-2. Verify in your app's logs that a similar message was posted:
+3. [Optional] Verify in your app's logs that a similar message was posted:
```
Comment (backwind1233) [inline servicebus/spring-cloud-azure-starter-servicebus-jms/servicebus-jms-queue/README.md:132]: same here

File: servicebus/spring-cloud-azure-starter-servicebus-jms/servicebus-jms-queue/src/main/java/com/azure/spring/sample/jms/queue/QueueSendController.java [hunk 7]
```diff
@@ -6,28 +6,34 @@
...
 import org.springframework.jms.core.JmsTemplate;
 import org.springframework.web.bind.annotation.PostMapping;
 import org.springframework.web.bind.annotation.RequestParam;
 import org.springframework.web.bind.annotation.RestController;
 
+/**
+ * Message Producer Mannually.
+ */
 @RestController
 public class QueueSendController {
```
[the remark this answers is not in the document]
Comment (backwind1233) [reply servicebus/spring-cloud-azure-starter-servicebus-jms/servicebus-jms-queue/src/main/java/com/azure/spring/sample/jms/queue/QueueSendController.java:19]: The Controller can be remained I think,  it has a post request.

**Comment by jialigit** [comment]
I should propose different pr for different issues. So this closed for now.

**Pull request closed without merging**